src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,121 @@
+import { ComponentPropsWithoutRef, useState, MouseEvent } from "react";
+import { FiChevronDown, FiChevronUp } from "react-icons/fi";
+
+interface IOption extends ComponentPropsWithoutRef<"div"> {
+  icon?: JSX.Element;
+  item: string;
+}
+
+interface IDropdownProps extends ComponentPropsWithoutRef<"div"> {
+  options: IOption[];
+  size?: "xs" | "md";
+}
+
+function Icon({ toggle }: { toggle: boolean }) {
+  let IconclassName = "w-6 h-6 stroke-current";
+
+  return toggle ? (
+    <FiChevronUp className={IconclassName} />
+  ) : (
+    <FiChevronDown className={IconclassName} />
+  );
+}
+
+export function Dropdown(props: IDropdownProps) {
+  let { size = "md", options } = props;
+
+  const classes = {
+    xs: {
+      button: "w-full px-3 py-4 h-7 rounded hover:rounded",
+      panel: "w-full rounded hover:rounded",
+      item: "px-3 py-4 h-7 rounded hover:rounded",
+      icon: "pr-3",
+    },
+    md: {
+      button: "w-full px-6 py-3 h-14 rounded-lg hover:rounded-lg",
+      panel: "w-full rounded-lg hover:rounded-lg",
+      item: "w-full px-6 py-3 h-14 rounded-lg hover:rounded-lg",
+      icon: "pr-2",
+    },
+  };
+
+  const firstObject = options[0];
+  const [toggle, setToggle] = useState(false);
```

Comment:
```suggestion
  const [toggle, setToggle] = useState<bool>(false);
```

I think it's better to precise the type for each var present in a useState. It's more clearly.

Reply:
Agree! However it does not work like that. We apply when we figure it out. WDYT!?